dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/DatadogCore.kt

```diff
@@ -64,7 +63,14 @@ internal class DatadogCore(
     internal var webViewRumFeature: SdkFeature<Any, Configuration.Feature.RUM>? = null
 
     // TODO RUMM-0000 handle context
-    internal var contextProvider: ContextProvider? = null
+    internal val contextProvider: ContextProvider?
```

Comment:
why not using the `NoOpContextProvider` here and make this non - optional ?

Reply:
maybe a bit later, because the lifecycle and context management are not yet final